packages/mcdoc/src/runtime/checker/index.ts

```diff
@@ -979,9 +975,17 @@ function simplifyIndexed<T>(
 				)
 		)
 		if (currentValues.includes(undefined)) {
-			// fallback case
-			values = child.fields.filter(f => f.kind === 'pair').map(f => f.type)
-			break
+			const fallbackDispatch = child.fields.find(f =>
+				f.kind === 'pair' && f.key.kind === 'literal' && f.key.value.value === '%fallback'
+			)
+
+			if (fallbackDispatch) {
+				values.push(fallbackDispatch.type)
```

Comment:
What are these changes supposed to achieve? Why was this necessary?

Reply:
Because otherwise, it creates a union of all dispatched types, rather than using the defined `%fallback`